src/RemoteInboxNotifications/DataSourcePoller.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Handles polling and storage of specs
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+use Automattic\WooCommerce\Admin\Loader;
+
+/**
+ * Specs data source poller class.
+ * This handles polling specs from JSON endpoints, and
+ * stores the specs in to the database as an option.
+ */
+class DataSourcePoller {
+	const DATA_SOURCES = array(
+		'http://two.wordpress.test/rinds-specs.json',
+	);
+
+	/**
+	 * Polls the data sources for specs, persists those specs, and delete any
+	 * specs that no longer exist in the data sources.
+	 */
+	public static function poll_data_sources() {
```

Comment:
I've called it `read_specs` because 1. it doesn't return anything so `get` isn't quite right and 2. it reads and persists all specs, not just looking for latest - specs can change.